From 1c25162b7357dcdcf2370e9c2ec37ed8b2c9feb8 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Wed, 22 May 2019 12:58:47 +0100
Subject: [PATCH 503/773] vc4: Remove interrupt and DMA trampling

As part of the effort to clean up the overlays, remove the interrupt
and DMA mask declarations from the vc4 overlays which just duplicate
that which is in the base DTBs.

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 .../boot/dts/overlays/vc4-fkms-v3d-overlay.dts |  8 --------
 .../boot/dts/overlays/vc4-kms-v3d-overlay.dts  | 18 ++----------------
 2 files changed, 2 insertions(+), 24 deletions(-)

--- a/arch/arm/boot/dts/overlays/vc4-fkms-v3d-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-fkms-v3d-overlay.dts
@@ -60,7 +60,6 @@
 	fragment@7 {
 		target = <&v3d>;
 		__overlay__  {
-			interrupts = <1 10>;
 			status = "okay";
 		};
 	};
@@ -72,13 +71,6 @@
 		};
 	};
 
-	fragment@9 {
-		target-path = "/soc/dma";
-		__overlay__ {
-			brcm,dma-channel-mask = <0x7f35>;
-		};
-	};
-
 	__overrides__ {
 		cma-256 = <0>,"+0-1-2-3-4";
 		cma-192 = <0>,"-0+1-2-3-4";
--- a/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
+++ b/arch/arm/boot/dts/overlays/vc4-kms-v3d-overlay.dts
@@ -62,7 +62,6 @@
 	fragment@7 {
 		target = <&pixelvalve0>;
 		__overlay__  {
-			interrupts = <2 13>; /* pwa0 */
 			status = "okay";
 		};
 	};
@@ -70,7 +69,6 @@
 	fragment@8 {
 		target = <&pixelvalve1>;
 		__overlay__  {
-			interrupts = <2 14>; /* pwa1 */
 			status = "okay";
 		};
 	};
@@ -78,7 +76,6 @@
 	fragment@9 {
 		target = <&pixelvalve2>;
 		__overlay__  {
-			interrupts = <2 10>; /* pixelvalve */
 			status = "okay";
 		};
 	};
@@ -86,7 +83,6 @@
 	fragment@10 {
 		target = <&hvs>;
 		__overlay__  {
-			interrupts = <2 1>;
 			status = "okay";
 		};
 	};
@@ -94,7 +90,6 @@
 	fragment@11 {
 		target = <&hdmi>;
 		__overlay__  {
-			interrupts = <2 8>, <2 9>;
 			status = "okay";
 		};
 	};
@@ -102,7 +97,6 @@
 	fragment@12 {
 		target = <&v3d>;
 		__overlay__  {
-			interrupts = <1 10>;
 			status = "okay";
 		};
 	};
@@ -115,14 +109,6 @@
 	};
 
 	fragment@14 {
-		target-path = "/soc/dma";
-		__overlay__ {
-			brcm,dma-channel-mask = <0x7f35>;
-		};
-	};
-
-
-	fragment@15 {
 		target = <&clocks>;
 		__overlay__  {
 			claim-clocks = <
@@ -134,14 +120,14 @@
 		};
 	};
 
-	fragment@16 {
+	fragment@15 {
 		target = <&vec>;
 		__overlay__  {
 			status = "okay";
 		};
 	};
 
-	fragment@17 {
+	fragment@16 {
 		target = <&txp>;
 		__overlay__  {
 			status = "okay";
